04-data-structures-part1.Rmd

```diff
@@ -43,6 +43,7 @@ is.numeric() # is it numeric?
 is.integer() # is it an integer?
 is.complex() # is it complex number data?
 is.character() # is it character data?
+str()  # what is it?
```

Comment:
Two thoughts:

1. Both when i read this and taught this, this seemed like an appropriate
place to introduce str(x). One could disagree. class() is not used at all
in this lesson.

2. The tutorial does use str(x) later, but without much explanation. So
this provides a prior opportunity to explain, and explain that there are
more general interrogation methods.



On Sun, Aug 23, 2015 at 9:34 PM, Naupaka Zimmerman <notifications@github.com
> wrote:

> In 04-data-structures-part1.Rmd
> <https://github.com/swcarpentry/r-novice-gapminder/pull/58#discussion_r37715661>
> :
>
> > @@ -43,6 +43,7 @@ is.numeric() # is it numeric?
> >  is.integer() # is it an integer?
> >  is.complex() # is it complex number data?
> >  is.character() # is it character data?
> > +str()  # what is it?
>
> There's also class() -- but I think of str() and class() as more general
> interrogation functions than just for atomic types and it might get a tad
> confusing. E.g.:
>
> > a <- 1> str(a)
>  num 1> class(a)
> [1] "numeric"> typeof(a)
> [1] "double"
>
> Is this the best place to have them introduced in the lesson? I definitely
> agree they should be in there somewhere though, and this indeed might be
> the best place to do it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/swcarpentry/r-novice-gapminder/pull/58/files#r37715661>
> .
>